src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,454 @@
+                transaction_type: str = Keyword.BUY.value if RP2Decimal(record[_AMOUNT]) > ZERO else Keyword.SELL.value
+
+                unique_id: str = Keyword.UNKNOWN.value if spot_price is Keyword.UNKNOWN.value else key
+
+                if RP2Decimal(record[_AMOUNT]) > ZERO:
+                    crypto_in: str = str(amount)
+                    fiat_in_no_fee: str = str(RP2Decimal(trade_history[record[_REFID]][_COST]) - RP2Decimal(
+                        trade_history[record[_REFID]][_FEE]))
+                    fiat_in_with_fee: str = trade_history[record[_REFID]][_COST]
+                    result.append(
+                        InTransaction(
+                            plugin=self.__PLUGIN_NAME,
+                            unique_id=unique_id,
+                            raw_data=raw_data,
+                            timestamp=timestamp_value,
+                            asset=asset_base,
+                            exchange=self.__EXCHANGE_NAME,
+                            holder=self.account_holder,
+                            transaction_type=transaction_type,
+                            spot_price=spot_price,
+                            crypto_in=crypto_in,
+                            crypto_fee=crypto_fee,
+                            fiat_in_no_fee=fiat_in_no_fee,
+                            fiat_in_with_fee=fiat_in_with_fee,
+                            fiat_fee=fiat_fee,
+                            notes=key,
+                        )
+                    )
+                else:
+                    crypto_out_no_fee: str = str(amount)
+                    crypto_out_with_fee: str = str(amount + RP2Decimal(record[_FEE]))
+                    fiat_out_no_fee: str = str(RP2Decimal(trade_history[record[_REFID]][_COST]) - RP2Decimal(
+                        trade_history[record[_REFID]][_FEE]))
+
+                    result.append(
+                        OutTransaction(
+                            plugin=self.__PLUGIN_NAME,
+                            unique_id=unique_id,
+                            raw_data=raw_data,
+                            timestamp=timestamp_value,
+                            asset=asset_base,
+                            exchange=self.__EXCHANGE_NAME,
+                            holder=self.account_holder,
+                            transaction_type=transaction_type,
+                            spot_price=spot_price,
+                            crypto_out_no_fee=crypto_out_no_fee,
+                            crypto_fee=crypto_fee,
+                            crypto_out_with_fee=crypto_out_with_fee,
+                            fiat_out_no_fee=fiat_out_no_fee,
+                            fiat_fee=fiat_fee,
+                            notes=key,
+                        )
+                    )
+            elif record[_TYPE] == _MARGIN or record[_TYPE] == _ROLLOVER:
```

Comment:
Nit. It's bit more Pythonic this way:
```suggestion
            if record[_TYPE] in {_MARGIN, _ROLLOVER}:
```